[SRVLOGIC-469]: Deploying workflows chapter does not use kn worklow plugin

domhanak opened the pull request:
Now it uses the kn workflow plugin

<!--- PR title format: [GH#<gh-issue-id>][BZ#<bz-issue-id>][OCPBUGS#<jira-issue-id>][OSDOCS#<jira-issue-id>]: <short-description-of-the-pr> --->

<!--- If your changes apply to the latest release and/or in-development version of OpenShift, open your PR against the `main` branch.
Do not create or rename a top-level directory (or any subdirectory in a directory that contains a hugebook.flag file) in the repository and topic map without checking with a docs program manager first.
If a book is being created or modified, there are changes on the Customer Portal that must also be made.

* For more details about the information requested in this template, see:
  https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/create_or_edit_content.adoc#submit-PR --->

Version(s):
<!--- Specify the version or versions of OpenShift your PR applies to. -->
`serverless-docs-1.34`
`serverless-docs-1.35`

Issue:
<!--- Add a link to the Bugzilla, Jira, or GitHub issue, if applicable. --->
[SRVLOGIC-269](https://issues.redhat.com/browse/SRVLOGIC-469)

Link to docs preview:
<!--- Add direct link(s) to the exact page(s) with updated content from the preview build. --->
[Preview](https://85916--ocpdocs-pr.netlify.app/openshift-serverless/latest/serverless-logic/serverless-logic-getting-started/serverless-logic-deploying-workflows.html)

QE review:
- [ ] QE has approved this change.
<!--- QE approval is required to merge a PR except for changes that do not impact the meaning of the docs. --->

Additional information:
<!--- Optional: Include additional context or expand the description here.--->

<!--- After you open your PR, ask for review from the OpenShift docs team:
  For community authors: Tag @openshift/team-documentation in a GitHub comment.--->
  
  @kaldesai @wmedvede @ricardozanini PTAL

domhanak committed:
SRVLOGIC-469: Deploying workflows chapter does not use kn worklow plugin

Now it uses the kn workflow plugin

domhanak commented on modules/serverless-logic-building-deploying-workflow-preview-mode.adoc:
```diff
@@ -12,70 +12,28 @@ You can create a `SonataFlow` custom resource (CR) on {ocp-product-title} and {S
 [source,terminal]
 ----
-$ oc apply -f greetings-workflow.yaml -n workflows
+$ kn workflow deploy --custom-manifests-dir=./manifests --namespace <your_namespace>
```
Since it is docs,  I wanted to use the explicit version of commands.

domhanak commented on modules/serverless-logic-building-deploying-workflow-preview-mode.adoc:
```diff
@@ -12,70 +12,28 @@ You can create a `SonataFlow` custom resource (CR) on {ocp-product-title} and {S
 [source,terminal]
 ----
-$ oc apply -f <workflow-name>.yaml -n <your_namespace>
+$ kn workflow gen-manifest --mode=preview --namespace <your_namespace>
```
Good catch should be:
```suggestion
$ kn workflow gen-manifest --profile=preview --namespace <your_namespace>
```

domhanak committed:
Fix incorrect flag used mode -> preview

domhanak commented:
Need to also fix the devmode and the other referenced guide. Ty for review, so far will ping you when I push the changes

kaldesai commented:
/ok-to-test

domhanak commented on modules/serverless-logic-building-deploying-workflow-preview-mode.adoc:
```diff
@@ -12,70 +12,28 @@ You can create a `SonataFlow` custom resource (CR) on {ocp-product-title} and {S
-$ oc apply -f greetings-workflow.yaml -n workflows
+$ kn workflow deploy --custom-manifests-dir=./manifests --namespace <your_namespace>
 ----
++
```
```suggestion

```

domhanak committed:
Update modules/serverless-logic-building-deploying-workflow-preview-mode.adoc

domhanak commented on modules/serverless-logic-building-deploying-workflow-preview-mode.adoc:
```diff
@@ -113,4 +70,13 @@ Ensure that the pod corresponding to your workflow is running.
+----
+$ kn workflow undeploy -n <your_namespace>
+----
++
```
```suggestion

```

domhanak committed:
Update modules/serverless-logic-building-deploying-workflow-preview-mode.adoc

kaldesai commented on modules/serverless-logic-building-deploying-workflow-preview-mode.adoc:
```diff
@@ -12,69 +12,26 @@ You can create a `SonataFlow` custom resource (CR) on {ocp-product-title} and {S
 ----
 +
-.Example command for the `greetings-workflow.yaml` file:
+This command generates {ocp-product-title} YAML manifests based on your `workflow.sw.json|yaml` definition. Default location is `<project_application_dir>/manifests` directory.
```
```suggestion
This command generates {ocp-product-title} YAML manifests based on your `workflow.sw.json|yaml` definition. The default location is the `<project_application_dir>/manifests` directory.
```

kaldesai commented on modules/serverless-logic-building-deploying-workflow-preview-mode.adoc:
```diff
@@ -12,69 +12,26 @@ You can create a `SonataFlow` custom resource (CR) on {ocp-product-title} and {S
 +
-.Example command for the `greetings-workflow.yaml` file:
+This command generates {ocp-product-title} YAML manifests based on your `workflow.sw.json|yaml` definition. Default location is `<project_application_dir>/manifests` directory.
+You can modify these files to your liking, however when you run above command again, the changes are going to be overwritten. Use `--custom-generated-manifests-dir` or `-c` flag to generate the files in different location.
```
```suggestion
You can modify these files to your preference, however, when you rerun the above command, the changes will be overwritten. Use the `--custom-generated-manifests-dir` or `-c` flag to generate the files in different locations.
```

kaldesai commented on modules/serverless-logic-deploying-workflows-dev-mode.adoc:
```diff
@@ -12,69 +12,32 @@ You can deploy your local workflow on {ocp-product-title} in Dev mode. You can u
 .Procedure
 
-. Create the workflow configuration YAML file.
+. Login to your cluster
```
```suggestion
. Login to your cluster.
```

kaldesai reviewed (approved):
A minor suggestions. Otherwise, LGTM!

kaldesai commented:
@domhanak Once you are done with drafting the content, you can assign this PR for the peer review process by adding a `/label peer-review-needed` label to this. 

Then the OpenShift peer review squad will peer review this PR and then you can proceed for the merge review process.

domhanak committed:
Apply suggestions from code review

Review from Kalyani

Co-authored-by: Kalyani Desai <43639538+kaldesai@users.noreply.github.com>

domhanak committed:
SRVLOGC-469: Reorganize Deploying Workflow doc

Reorder docs to make more sense and follow the precedures
from basic (what user should do 1st, to more complex ones )

domhanak committed:
SRVLOGIC-469: Improve verifying-workflow-deployment-preview-mode

Removes the example workflow from the module and replaces it with
prerequsites that user should have the WF deployed.
Adds steps to query all services.

domhanak committed:
SRVLOGIC-469: Fix typos and rebase

domhanak committed:
SRVLOGIC-469: Fix typos introducing formating issues

domhanak committed:
SRVLOGIC-469: Fix additonal typos, use <namespace> instead of workflows

kaldesai commented:
@domhanak You want me to put this on the peer review as well?

domhanak committed:
SRVLOGIC-469: Improve deploying dev and preview mode docs

Unify namespace usage.
Use shorthand -n where possible.
Add Testing section to deploying-workflows-dev-mode

domhanak commented:
@kaldesai not yet I still feel like some parts can be enhanced. I ll ping you :)

domhanak committed:
SRVLOGIC-469: Improve deploying-workflows-dev-mode Testing section

kaldesai commented:
@domhanak Do you want me to put this on the peer review as well?

kaldesai commented:
/label peer-review-needed

adellape commented on modules/serverless-logic-building-deploying-workflow-preview-mode.adoc:
```diff
@@ -12,89 +12,46 @@ You can create a `SonataFlow` custom resource (CR) on {ocp-product-title} and {S
 ----
 +
-.Example command for the `greetings-workflow.yaml` file:
+.Example command for a workflow that has ID attribute with value `greeting`:
```
No colons for titles:
```suggestion
.Example command for a workflow that has ID attribute with value `greeting`
```

adellape commented on modules/serverless-logic-building-deploying-workflow-preview-mode.adoc:
```diff
@@ -112,5 +69,13 @@ Ensure that the pod corresponding to your workflow is running.
+$ oc logs pod/<pod-name> -n <namespace>
+----
+
+.Cleanup
```
Technically `.Cleanup` is a non-standard heading in a procedure module (per [modular docs guide](https://redhat-documentation.github.io/modular-docs/#con-creating-procedure-modules_writing-mod-docs)). FWIW, I see only 1 other instance in the repo (that one has it as `.Clean up` two-words, tho the single-word here is correctly what IBM SG suggests).

Could this instead just be the last step in the normal `.Procedure`? Or maybe the last step in `.Verification`? Alternatively, could be its own procedure "Cleaning up a workflow deployment" but that seems like overkill.

adellape commented on modules/serverless-logic-building-deploying-workflow-preview-mode.adoc:
```diff
@@ -112,5 +69,13 @@ Ensure that the pod corresponding to your workflow is running.
+----
+
+.Cleanup
+. To cleanup the deployment, delete the projects resources running the following command:
```
Either way though, If this stays as a single-step procedure, use an unordered list (per [guidelines](https://redhat-documentation.github.io/modular-docs/#con-creating-procedure-modules_writing-mod-docs)) and use "by running":
```suggestion
* To cleanup the deployment, delete the projects resources by running the following command:
```

adellape commented on modules/serverless-logic-deploying-workflows-dev-mode.adoc:
```diff
@@ -12,75 +12,39 @@ You can deploy your local workflow on {ocp-product-title} in Dev mode. You can u
-. To deploy the application, apply the YAML file by entering the following command:
+. In your terminal, navigate to the root directory of your {ServerlessLogicProductName} project.
+
+. Deploy the application, by running the following command:
```
```suggestion
. Deploy the application by running the following command:
```

adellape commented on modules/serverless-logic-deploying-workflows-dev-mode.adoc:
```diff
@@ -102,23 +66,48 @@ $ oc edit sonataflow <workflow_name> -n <your_namespace>
+$ oc logs <workflow_pod_name> -n <namespace>
 ----
 
+.Testing
```
Same thought re: non-standard heading for this `.Testing` one. This one feels like splitting it out as its own procedure would be worthwhile (versus including it within another procedure module).

Edit: I see now there already is a ["Testing a deployment"](https://85916--ocpdocs-pr.netlify.app/openshift-serverless/latest/serverless-logic/serverless-logic-getting-started/serverless-logic-deploying-workflows.html#serverless-logic-testing-workflows_serverless-logic-deploying-workflows) procedure later in the assembly. Is that basically the same procedure that's being shown here? If so, could there just be an "Additional resources" link to that procedure from this one instead of duplicating?

adellape commented on modules/serverless-logic-deploying-workflows-dev-mode.adoc:
```diff
@@ -102,23 +66,48 @@ $ oc edit sonataflow <workflow_name> -n <your_namespace>
+$ oc get route/<workflow_name> -n <namespace>
+----
++
+Note the value of `HOST/PORT` from the result of tis command, referenced as `<WORKFLOW_URL>` in next step.
```
Typo x2:
```suggestion
Note the value of `HOST/PORT` from the result of this command, referenced as `<WORKFLOW_URL>` in the next step.
```
Also, if `<WORKFLOW_URL>` only appears as part of this documentation / procedure, could it be lowercase like a normal user-replaceable? E.g.:
```suggestion
Note the value of `HOST/PORT` from the result of this command, referenced as `<workflow_url>` in the next step.
```
If it has meaning as all-cap outside of this procedure, however (like in output or the UI someplace), then ignore.

adellape commented on modules/serverless-logic-deploying-workflows-dev-mode.adoc:
```diff
@@ -102,23 +66,48 @@ $ oc edit sonataflow <workflow_name> -n <your_namespace>
++
+Note the value of `HOST/PORT` from the result of tis command, referenced as `<WORKFLOW_URL>` in next step.
+
+.. Navigate to the `<WORKFLOW_URL>/q/dev-ui/org.apache.kie.sonataflow.sonataflow-quarkus-devui/workflows` in your browser. Please note that it may take some time until the management console is available. 
```
Re: previous comment, match lowercase here as well, if you change the other:
```suggestion
.. Navigate to the `<workflow_url>/q/dev-ui/org.apache.kie.sonataflow.sonataflow-quarkus-devui/workflows` in your browser. Please note that it may take some time until the management console is available. 
```

adellape replied:
* Don't need "please", or maybe even "note".
* "might" instead of "may" (better for translation).
```suggestion
.. Navigate to the `<WORKFLOW_URL>/q/dev-ui/org.apache.kie.sonataflow.sonataflow-quarkus-devui/workflows` in your browser. It might take some time until the management console is available. 
```

adellape commented on modules/serverless-logic-deploying-workflows-dev-mode.adoc:
```diff
@@ -102,23 +66,48 @@ $ oc edit sonataflow <workflow_name> -n <your_namespace>
+.. Navigate to the `<WORKFLOW_URL>/q/dev-ui/org.apache.kie.sonataflow.sonataflow-quarkus-devui/workflows` in your browser. Please note that it may take some time until the management console is available. 
+
+
 .Next steps
```
This single-step `.Next steps` procedure could probably just be absorbed into wherever the `.Testing` procedure above it ends up.

adellape commented on modules/serverless-logic-verifying-workflow-deployment-preview-mode.adoc:
```diff
@@ -13,62 +13,22 @@ You can verify that your {ServerlessLogicProductName} workflow is running by per
 * You have an {ServerlessLogicOperatorName} installed on your cluster.
 * You have access to an {ServerlessLogicProductName} project with the appropriate roles and permissions to create applications and other workloads in {ocp-product-title}.
 * You have installed the OpenShift CLI `(oc)`.
+* You have deployed your workflow in preview mode in the {ocp-product-title}.
```
Drop "the":
```suggestion
* You have deployed your workflow in preview mode in {ocp-product-title}.
```

adellape commented on modules/serverless-logic-verifying-workflow-deployment-preview-mode.adoc:
```diff
@@ -13,62 +13,22 @@ You can verify that your {ServerlessLogicProductName} workflow is running by per
 .Procedure
 
-. Create a workflow `YAML` file similar to the following:
+. Retrieve the list of service and identify the one matching `name` of your workflow:
```
Should this be plural?
```suggestion
. Retrieve the list of services and identify the one matching the `name` of your workflow:
```

domhanak commented:
@adellape You made me think hard about the guide and how it looks as whole I am trying to improve it
